maze.pl

```diff
@@ -136,9 +151,24 @@
 % Ensure that the tests pass with no warnings.
 %
 
+path_to(From,To,Path) :- connected(From,To); connected(To,From);
+                         connected(To,Path);
+			 connected(From,Path),!.
+
+
+path_to(X,Y,[X,Y]):-corridor(X,Y),!.
+path_to(X,Y,[X|R]):-corridor(X,Z),path_to(Z,Y,R),!.
+
```

Comment:
path_to, while getting the correct result, still doesn't cut properly, leaving a choice point in the tests.  This is probably from the first rule (line 154).